internal/poller/poller.go

```diff
@@ -0,0 +1,97 @@
+//go:generate mockgen -source=poller.go -destination=mocks/poller.go -package=mocks
+package poller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rudderlabs/rudder-control-plane-sdk/modelv2"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+// Poller periodically polls for new workspace configs and runs a handler on them.
+type Poller struct {
+	client    Client
+	interval  time.Duration
+	handler   WorkspaceConfigHandler
+	updatedAt time.Time
+	log       logger.Logger
+}
+
+type WorkspaceConfigHandler func(*modelv2.WorkspaceConfigs) error
+
+type Client interface {
+	GetWorkspaceConfigs(ctx context.Context) (*modelv2.WorkspaceConfigs, error)
+	GetUpdatedWorkspaceConfigs(ctx context.Context, updatedAt time.Time) (*modelv2.WorkspaceConfigs, error)
+}
+
+func New(handler WorkspaceConfigHandler, opts ...Option) (*Poller, error) {
+	p := &Poller{
+		interval: 1 * time.Second,
+		handler:  handler,
+		log:      logger.NOP,
+	}
+
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	if p.handler == nil {
+		return nil, fmt.Errorf("handler is required")
+	}
+
+	if p.client == nil {
+		return nil, fmt.Errorf("client is required")
+	}
+
+	return p, nil
+}
+
+// Start starts the poller goroutine. It will poll for new workspace configs every interval.
+// It will stop polling when the context is cancelled.
+func (p *Poller) Start(ctx context.Context) {
```

Comment:
[minor] Although we have a way to stop this goroutine, there is no way to know when it got terminated or not. This could be an issue in some cases.

I would suggest on of the following:
1. Make `Start` a blocking method, called `Run(ctx context.Context)`
2. Introduce a background context, remove ctx from params and add a `Stop` method, that blocks until the go routine returns